test/built-ins/Promise/prototype/then/ctor-throws.js

```diff
@@ -0,0 +1,39 @@
+// Copyright (C) 2016 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+description: >
+    `Promise.prototype.then` invoked on a constructor value that throws an
+    error
+es6id: 25.4.5.3
+info: >
+    1. Let promise be the this value.
+    [...]
+    3. Let C be SpeciesConstructor(promise, %Promise%).
+    [...]
+    5. Let resultCapability be NewPromiseCapability(C).
+    6. ReturnIfAbrupt(resultCapability).
+
+    25.4.1.5 NewPromiseCapability
+    [...]
+    6. Let promise be Construct(C, «executor»).
+    7. ReturnIfAbrupt(promise).
+    features: [Symbol.species]
+---*/
+
+var BadCtor = function() {
+  throw new Test262Error();
+};
+var originalSpecies = Object.getOwnPropertyDescriptor(Promise, Symbol.species);
+
+Object.defineProperty(Promise, Symbol.species, { value: BadCtor });
```

Comment:
Although maybe pre-mature their was discussion in removing this feature: https://github.com/tc39/tc39-notes/blob/master/es7/2015-11/nov-18.md#conclusionresolution-2